patch_recurrence/dehaze.py

```diff
@@ -77,13 +88,34 @@ def main():
     logger.info("Filtering pairs for checking normalized correlation ...")
     pairs = steps.filter_pairs(patches, pairs, constants)
 
+    print("\nNumber of pairs retained after filtering")
+    print(len(pairs))
+
     logger.info("Removing outliers ...")
     pairs = steps.remove_outliers(pairs, constants)
 
+    print("\nNumber of pairs retained after removing outliers")
+    print(len(pairs))
+
     logger.info("Estimating global airlight ...")
     airlight = steps.estimate_airlight(pairs)
 
     logger.info("Estimated airlight is ...%s", str(airlight))
 
+    # T-map estimation code begins
+
+    del pairs
+    del patches
+    gc.collect()
```

Comment:
Yes it definitely does. But I still think that instead of re-obtaining all patches, I should try to get them all at once in the first step itself. I should then make use of alternate patches for the initial steps and later make use of all of them for t-map estimation. I will try to remove this unnecessary repetition.